File: 0901-online-stock-span/0901-online-stock-span.py
class StockSpanner:

    def __init__(self):
        self.stack=[]
        self.Arr=[]
        self.i=-1
        

    def next(self, price: int) -> int:
        self.Arr.append(price)
        self.i+=1
        flag=0
        while(self.stack and self.Arr[self.stack[-1]]<=price):
            self.stack.pop()
            flag=1
        self.stack.append(self.i)
        if flag==0:
            return 1
        else:
            if len(self.stack)==1:
                return self.stack[-1]+1
            else:
                return self.stack[-1]-self.stack[-2]
                
        
                   
            
            
        


# Your StockSpanner object will be instantiated and called as such:
# obj = StockSpanner()
# param_1 = obj.next(price)

File: 0901-online-stock-span/0901-online-stock-span.cpp
class StockSpanner {
public:
    vector<int>stockprices;
    vector<int>span;
    int i;
    StockSpanner() {
        i=-1;
    }
    
    int next(int price) {
        int res=1;
        stockprices.push_back(price);
        i++;
        if(span.size()==0)
        {
            span.push_back(i);
            return res;
        }
        while(span.size()>0 and stockprices[span.back()]<=price)
        {
            span.pop_back();
        }
        if(span.size()==0)res=stockprices.size();
        else res=stockprices.size()-1-span.back();
        span.push_back(i);
        return res;
        
    }
};

/**
 * Your StockSpanner object will be instantiated and called as such:
 * StockSpanner* obj = new StockSpanner();
 * int param_1 = obj->next(price);
 */